sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -929,81 +929,95 @@ def sdc_pandas_dataframe_drop_impl(df, _func_name, args, columns):
     return sdc_pandas_dataframe_drop_impl(df, _func_name, args, columns)
 
 
-def df_getitem_str_literal_idx_main_codelines(self, idx):
+def df_getitem_slice_idx_main_codelines(self, idx):
     """Generate main code lines for df.getitem"""
-    try:
-        col_idx = self.columns.index(idx)
-    except ValueError:
-        func_lines = ['  raise KeyError']
+    if isinstance(self.index, types.NoneType):
+        func_lines = ['  length = len(get_dataframe_data(self, 0))',
+                      '  index = numpy.arange(length)']
     else:
-        col_data = f'data_{col_idx}'
-        func_lines = [
-            f'  {col_data} = get_dataframe_data(self, {col_idx})',
-            f'  return pandas.Series({col_data}, index=self._index, name=idx)'
+        func_lines = ['  index = self._index']
+
+    results = []
+    for i, col in enumerate(self.columns):
+        res_data = f'result_data_{i}'
+        func_lines += [
+            f'  data_{i} = get_dataframe_data(self, {i})',
+            f'  {res_data} = pandas.Series(data_{i}[idx], index=index[idx], name="{col}")'
```

Comment:
It might be better to compute index[idx] once and just use it in these lines, to avoid repetitive calls.